src/EditorFeatures/CSharpTest/Classification/SemanticClassifierTests.cs

```diff
@@ -2182,7 +2182,7 @@ void goo()
                 WpfTestCase.RequireWpfFact("Creates an IWpfTextView explicitly with an unrelated buffer");
                 using (var disposableView = workspace.ExportProvider.GetExportedValue<ITextEditorFactoryService>().CreateDisposableTextView(extraBuffer))
                 {
-                    var listenerProvider = new AsynchronousOperationListenerProvider();
+                    var listenerProvider = workspace.ExportProvider.GetExportedValue<IAsynchronousOperationListenerProvider>();
```

Comment:
tagging @dotnet/roslyn-ide @jinujoseph @DustinCampbell @Pilchie @jasonmalinowski 

I think if we go this route, we should change our code to use [Import] property/fields rather than doing [ImportingConstructor]

we choose ImportingConstructor at the time with completely opposite mindset. we wanted to give in Mock to dependencies. and make test more API level unit tests.

we start to move to more behavior test kind of unit tests. start to use MEF to provide dependencies (since our dependency was too complex, almost impossible to provide all those mocks at right layer).

now with this change, where we use MEF provider itself as isolation point, I think ImportingConstructor actually work against what we want. since user can provide one at the constructor, and deep down other service getting same dependency from MEF export and messing up everything.

we should ban using ImportingConstructor from our code, and use [Import] rather to force, all dependencies are coming from MEF export.

Reply:
> rather than doing [ImportingConstructor]

We need to keep `[ImportingConstructor]` so we can block direct construction of the object (otherwise the default constructor would be public and accessible). The placement of dependencies as parameters vs. properties at that point makes little difference.

Reply:
hmm.. I dont understand what you mean by blocking direct construction? you can give in null to all dependencies. or mock to it. isn't one of point of what you are doing here is to make sure all dependencies get same services rather than different one?

why use code review to make sure it gets same dependency when you can make sure in does by the system?

having [Import] as same as I create object giving null to all dependencies?

Reply:
The goal of the change is for the following to be a *compilation error*:

```csharp
new AsynchronousOperationListenerProvider(...)
```

Alternatives to what I have implemented here are fine as long as that characteristic is preserved.

Reply:
that's interesting. so planning to mark all importing constructor as obsolete? that should work as well. 

so, when someone wants to test service itself like 
http://source.roslyn.io/#Roslyn.Services.Editor2.UnitTests/GoToDefinition/GoToDefinitionCommandHandlerTests.vb,103

they should get CSharpGotoDefinitionService from MEF from the beginning? ya. that should work as well rather than changing all code to [Import] from [ImportingConstructor]

so I guess that work is not done yet? I see only Async Listener Provider is marked obsolete. that can fix the problem where 2 different instance of IAsynchornounsListenerProvider is used. one directly given and one from MEF. but what about all other services?

Reply:
Correct, I have only made the change for two services to date (one in #25558, and one here).